src/foundation/contractible-types.lagda.md

```diff
@@ -298,3 +302,21 @@ module _
   pr1 (equiv-diagonal-is-contr X H) = const A X
   pr2 (equiv-diagonal-is-contr X H) = is-equiv-diagonal-is-contr H X
 ```
+
+### Contractiblity of iterated Σ-types
+
+```agda
+
+is-contr-Σ-telescope : {l : Level} {n : ℕ} → telescope l n → UU l
+is-contr-Σ-telescope (base-telescope A) = is-contr A
+is-contr-Σ-telescope (cons-telescope A) =
+  (is-contr X) × (Σ X (λ x → is-contr-Σ-telescope (A x)))
+  where X = _
```

Comment:
This was the best way I could find to recover `X` when I need it and it cannot be inferred, but as you can see it's quite awkward.

Reply:
It is inferred because of the Sigma, and then I can also use it for the contractibility proof